src/app/core/actions/transform/TranslateAction.ts

```diff
@@ -37,11 +39,13 @@ export class TranslateAction implements Action {
      * @param initialPositions Initializes the array with the selected components' starting positions
      * @param targetPositions Initializes the array with the selected components' final positions
      */
-    public constructor(objs: Component[], initialPositions: Vector[], targetPositions: Vector[]) {
+    public constructor(objs: Component[], initialPositions: Vector[], targetPositions: Vector[], snap?: boolean) {
         this.objs = objs;
 
         this.initialPositions = initialPositions;
         this.targetPositions = targetPositions;
+        if(typeof snap === "undefined"){ this.snap = true; }
+        else{ this.snap = snap;}
```

Comment:
This syntax can be cleaned up by changing `snap?: boolean` to `snap = true`. Then just `this.snap = snap;` is all that is necessary below.